tests/data/generator/two_party_negotiation/message.py

```diff
@@ -0,0 +1,91 @@
+        if type(other) is type(self):
+            raise NotImplementedError
+        else:
+            return False
+
+
+class TwoPartyNegotiationMessage(Message):
+    """A protocol for negotiation over a fixed set of resources involving two parties."""
+
+    def __init__(self, message_id: int, target: int, performative: str, contents: List, **kwargs):
+        """Initialise."""
+        super().__init__(message_id=message_id, target=target, performative=performative, contents=contents, **kwargs)
+
+        self.speech_acts = {
+            'cfp': {
+                'query', DataModel},
+            'propose': {
+                'query', DataModel,
+                'price', float},
+            'accept': {},
+            'decline': {},
+            'match_accept': {}}
+
+        assert self.check_consistency()
+
+    @property
+    def performatives(self) -> set:
+        """Get allowed performatives."""
+        performatives_set = set()
+        for performative in self.speech_acts:
+            performatives_set.add(performative)
+        return performatives_set
+
+    def check_consistency(self) -> bool:
+        """Check that the message follows the two_party_negotiation protocol."""
+        try:
+            assert self.is_set("message_id"), "message_id is not set"
+            message_id = self.get("message_id")
+            assert type(message_id) == int, "message_id is not int"
+
+            assert self.is_set("target"), "target is not set"
+            target = self.get("target")
+            assert type(target) == int, "target is not int"
+
+            assert self.is_set("performative"), "performative is not set"
+            performative = self.get("performative")
+            assert type(performative) == str, "performative is not str"
+
+            assert self.is_set("contents"), "contents is not set"
+            contents = self.get("contents")
+            assert type(contents) == list, "contents is not list"
+            contents = cast(List, contents)
+
+            # Light Protocol 2
+            # Check correct performative
+            assert performative in self.performatives, "performative is not in the list of allowed performative"
+
+            # Check correct contents
+            content_sequence_definition = self.speech_acts[performative]  # type is List
+            # Check number of contents
+            assert len(contents) == len(content_sequence_definition), "incorrect number of contents"
+            # Check the content is of the correct type
+            for content in range(len(content_sequence_definition)):
+                assert isinstance(contents[content], content_sequence_definition[content][1]), "incorrect content type"
+
+            # Light Protocol 3
+            if message_id == 1:
+                assert target == 0, "target should be 0"
+            else:
+                assert 1 < target < message_id, "target should be between 1 and message_id"
```

Comment:
assert 0 < target < message_id, "target should be strictly between 0 and message_id"

Reply:
Only the first message's target can be 0 (i.e. targets nothing since first move). 
The other moves must target an existing one (i.e. 1 < target < message_id).

Reply:
Yes, but your code excludes `target=1` which is a valid target for a message_id > 1